test/unit/org/apache/cassandra/cql3/validation/operations/UpdateTest.java

```diff
@@ -189,6 +191,34 @@ private void testUpdate(boolean forceFlush) throws Throwable
         }
     }
 
+    @Test
+    public void testUpdateWithContains() throws Throwable
+    {
+        createTable("CREATE TABLE %s (a int, b frozen<map<int, int>>, c int, primary key (a, b))");
+        Map<Integer, Integer> testMap = new HashMap<>();
+        testMap.put(1, 1);
+        testMap.put(2, 2);
+        execute("INSERT INTO %s (a, b, c) VALUES (?, ?, ?)", 1, testMap, 3);
+        assertRows(execute("SELECT * FROM %s"),
+                   row(1, testMap, 3));
+        assertInvalidMessage("Cannot use UPDATE with CONTAINS",
+                             "UPDATE %s SET c=3 WHERE a=1 AND b CONTAINS 1");
+    }
+
+    @Test
+    public void testUpdateWithContainsKey() throws Throwable
+    {
+        createTable("CREATE TABLE %s (a int, b frozen<map<int, int>>, c int, primary key (a, b))");
+        Map<Integer, Integer> testMap = new HashMap<>();
+        testMap.put(1, 1);
+        testMap.put(2, 2);
+        execute("INSERT INTO %s (a, b, c) VALUES (?, ?, ?)", 1, testMap, 3);
+        assertRows(execute("SELECT * FROM %s"),
+                   row(1, testMap, 3));
+        assertInvalidMessage("Cannot use UPDATE with CONTAINS KEY",
+                             "UPDATE %s SET c=3 WHERE a=1 AND b CONTAINS KEY 1");
+    }
+
```

Comment:
I think that we could merge the 2 tests into one. 
As each test requires cleaning up the tables and other slow operations having both tests together will minimize the amount of time spend to run the tests. 
Same thing for the DELETE tests.